src/PostgREST/Main.hs

```diff
@@ -71,6 +67,10 @@ main = do
         fail "Cannot run in this PostgreSQL version, PostgREST needs at least 9.2.0"
     ) supportedOrError
 
+  Right authenticator <- H.session pool $ do
```

Comment:
Actually don't worry about the error display part. I've got a patch locally that I can apply on top when this gets merged.